pkg/fission-cli/cmd/spec/spec.go

```diff
@@ -162,8 +163,8 @@ func save(data []byte, specDir string, specFile string) error {
 	return nil
 }
 
-// called from `fission * create --spec`
-func SpecSave(resource interface{}, specFile string) error {
+// called from `fission * create --spec` or `fission * create --dry`
+func SpecSave(resource interface{}, specFile string, input cli.Input) error {
```

Comment:
Hi @anubhav6663 , 

Sorry, maybe I didn't explain it well. I mean to move the encoding part to an independent function instead of merging two functions. The reason not to merge them is it makes harder for writing unit test.

Please check this sample code: https://gist.github.com/life1347/829271ea0dfa35e0ad05f18bba4b65ce